lib/ansible/modules/cloud/amazon/aws_ssm_send_command.py

```diff
@@ -0,0 +1,212 @@
+
+
+def main():
+    argument_spec = dict(
+        name=dict(required=True),
+        wait=dict(default=True, type='bool'),
+        comment=dict(),
+        instance_ids=dict(required=True, type='list'),
+        parameters=dict(default={}, type='dict')
+    )
+    module = AnsibleAWSModule(
+        argument_spec=argument_spec,
+        supports_check_mode=False
+    )
+
+    # Needs to be an existing SSM document name
+    document_name = module.params.get('name')
+    comment = module.params.get('comment')
+    await_return = module.params.get('wait')
+    instance_ids = module.params.get('instance_ids')
+    parameters = module.params.get('parameters')
+
+    if not (document_name and instance_ids):
+        module.fail_json(
+            msg="Must provide SSM document name and at least one instance id.")
+
+    conn = module.client('ssm')
+
+    invoke_params = {}
+
+    if document_name:
+        invoke_params['DocumentName'] = document_name
+    if comment:
+        invoke_params['Comment'] = comment
+    if instance_ids:
+        invoke_params['InstanceIds'] = instance_ids
+    if parameters:
+        invoke_params['Parameters'] = parameters
+
+    try:
+        response = ssm_send_command(conn, **invoke_params)
+    except botocore.exceptions.ClientError as ce:
+        if ce.response['Error']['Code'] == 'ResourceNotFoundException':
+            module.fail_json_aws(ce, msg="Could not find the SSM doc to execute. Make sure "
+                                 "the document name is correct and your profile has "
+                                 "permissions to execute SSM.")
+        module.fail_json_aws(
+            ce, msg="Client-side error when invoking SSM, check inputs and specific error")
+    except botocore.exceptions.ParamValidationError as ve:
+        module.fail_json_aws(
+            ve, msg="Parameters to `invoke` failed to validate")
+    except Exception as e:
+        module.fail_json(msg="Unexpected failure while invoking SSM send command.",
+                         exception=traceback.format_exc(e))
+
+    if await_return:
+        command_id = response['Command']['CommandId']
+        list_params = {}
+        if command_id:
+            list_params['CommandId'] = command_id
+            list_params['Details'] = True
+            checking = True
+            while checking:
+                try:
+                    invoke_response = ssm_list_command_invocations(
+                        conn, **list_params)
+                except Exception as e:
+                    module.fail_json(msg="Error in checking on execution status",
+                                     exception=traceback.format_exc(e))
+                if not invoke_response['CommandInvocations'] == []:
+                    if invoke_response['CommandInvocations'][0]['Status'] == 'Success':
+                        checking = False
+                    if invoke_response['CommandInvocations'][0]['Status'] == 'Failed':
```

Comment:
Replace this to something like:
`if invoke_response['CommandInvocations'][0]['Status'] in ['Failed', 'TimedOut', 'Cancelled']:`
to prevent an infinite loop from occurring when an invocation times out or is cancelled.